rinad/src/ipcm/ipcm.cc

```diff
@@ -1769,4 +1786,266 @@ void IPCManager_::io_loop(){
 
 }
 
+CatalogPsInfo::CatalogPsInfo(const rina::PsInfo& psinfo,
+			     map<string, CatalogPlugin>::iterator plit)
+				: PsInfo(psinfo)
+{
+	plugin = plit;
+}
+
+static bool endswith(const string& s, const string& suffix)
+{
+	if (s.size() < suffix.size()) {
+		return false;
+	}
+
+	return suffix == s.substr(s.size() - suffix.size(), suffix.size());
+}
+
+#define MANIFEST_SUFFIX ".manifest"
+
+void Catalog::import()
+{
+	const rinad::RINAConfiguration& config = IPCManager->getConfig();
```

Comment:
Instead of a back-reference to the IPCManager and have to do call `getConfig()`, this could be part of the `Catalogue` constructor